outputs.tf

```diff
@@ -41,7 +41,7 @@ output "service_credentials_object" {
 
 output "hostname" {
   description = "Database hostname. Only contains value when var.service_credential_names or var.users are set."
-  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.hostname"]) : length(var.users) > 0 ? data.ibm_database_connection.database_connection[0].https[0].hosts[0].hostname : null
+  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.hostname"]) : length(var.users) > 0 ? nonsensitive(flatten(data.ibm_database_connection.database_connection[0].https[0].hosts[0].hostname)) : null
```

Comment:
Please revert this change - you are re-introducing a bug here. The code in main branch is fine